src/main/java/jenkins/scm/api/SCMNavigatorDescriptor.java

```diff
@@ -84,7 +84,9 @@ public String getIconFilePathPattern() {
     /**
      * Creates a default navigator, if there should be one displayed.
      * May be interpreted by {@code jenkins.branch.SpecificDescriptor}.
-     * If returning non-null, you may also provide a {@code newInstanceDetail} view as per {@link TopLevelItemDescriptor}.
+     * If returning non-null, you may also provide a {@code newInstanceDetail} view as per {@link TopLevelItemDescriptor}
+     * and implement {@link SCMNavigatorDescriptor#getDescription()}.
```

Comment:
I think what you meant was that if you are returning non-null from `newInstance`, you should _either_ override `getDescription`, _or_ provide a `newInstanceDetail` with a more specialized appearance.